lib/response.js

```diff
@@ -53,6 +55,7 @@ internals.Response.prototype.writeHead = function () {
 
 internals.Response.prototype.write = function (data, encoding) {
 
+    this.shot.payloadChunks.push(new Buffer(data, encoding));
```

Comment:
It does bother me a little bit that the `ServerResponse` will also keep its own buffer (as long as it thinks it doesn't have a socket), so that data will be duplicated in memory.